crates/engine/tree/src/tree/mod.rs

```diff
@@ -1791,11 +1793,28 @@ where
         let hashed_state = self.provider.hashed_post_state(execution_output.state());
         let trie_updates = self.provider.get_block_trie_updates(block.number())?;
 
+        // Build trie data from the DB response synchronously and wrap it as a ready deferred
+        // handle.
+        let prefix_sets = hashed_state.construct_prefix_sets();
+        let sorted_hashed_state = Arc::new(hashed_state.into_sorted());
+        let sorted_trie_updates = Arc::new(trie_updates);
+        let trie_input = Arc::new(TrieInputSorted::new(
+            Arc::clone(&sorted_trie_updates),
+            Arc::clone(&sorted_hashed_state),
+            prefix_sets,
+        ));
+        let trie_data = ComputedTrieData {
+            hashed_state: sorted_hashed_state,
+            trie_updates: sorted_trie_updates,
+            anchor_hash: block.parent_hash(),
+            trie_input,
```

Comment:
I think given this use-case it might make sense to stick anchor_hash and trie_input inside a shared Option, so we can simply opt to not generate them for cases like this where it doesn't make sense to.